File: NovaMartPOS.Domain/Entities/Product.cs
namespace NovaMartPOS.Domain.Entities;

public class Product : BaseEntity
{
    public string ProductCode { get; set; } = string.Empty;
    public string Barcode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public decimal PurchasePrice { get; set; }
    public decimal SellingPrice { get; set; }
    public decimal TaxRate { get; set; }
    public int StockQuantity { get; set; }
    public int MinimumStockLevel { get; set; }
    public bool IsActive { get; set; } = true;

    public int CategoryId { get; set; }
    public Category Category { get; set; } = null!;

    public int BrandId { get; set; }
    public Brand Brand { get; set; } = null!;

    public int? SupplierId { get; set; }
    public Supplier? Supplier { get; set; }
}

File: nova-mart-frontend/src/types/product.ts
export interface Product {
    id: number;
    productCode: string;
    barcode: string;
    name: string;
    purchasePrice: number;
    sellingPrice: number;
    taxRate: number;
    stockQuantity: number;
    minimumStockLevel: number;
    isActive: boolean;
    categoryId: number;
    categoryName: string;
    brandId: number;
    brandName: string;
    supplierId: number | null;
  }
  
  export interface CreateProductRequest {
    productCode: string;
    barcode: string;
    name: string;
    purchasePrice: number;
    sellingPrice: number;
    taxRate: number;
    stockQuantity: number;
    minimumStockLevel: number;
    categoryId: number;
    brandId: number;
    supplierId?: number | null;
  }
  
  export interface UpdateProductRequest extends CreateProductRequest {
    isActive: boolean;
  }